sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/physical/partitioning.scala

```diff
@@ -694,28 +694,32 @@ case class KeyGroupedShuffleSpec(
     //        transform functions.
     //  4. the partition values, if present on both sides, are following the same order.
     case otherSpec @ KeyGroupedShuffleSpec(otherPartitioning, otherDistribution) =>
-      val expressions = partitioning.expressions
-      val otherExpressions = otherPartitioning.expressions
-
       distribution.clustering.length == otherDistribution.clustering.length &&
-        numPartitions == other.numPartitions &&
-          expressions.length == otherExpressions.length && {
-            val otherKeyPositions = otherSpec.keyPositions
-            keyPositions.zip(otherKeyPositions).forall { case (left, right) =>
-              left.intersect(right).nonEmpty
-            }
-          } && expressions.zip(otherExpressions).forall {
-            case (l, r) => isExpressionCompatible(l, r)
-          } && partitioning.partitionValuesOpt.zip(otherPartitioning.partitionValuesOpt).forall {
+        numPartitions == other.numPartitions && isExpressionsCompatible(otherSpec) &&
+          partitioning.partitionValuesOpt.zip(otherPartitioning.partitionValuesOpt).forall {
             case (left, right) => left.zip(right).forall { case (l, r) =>
               ordering.compare(l, r) == 0
             }
-         }
+          }
     case ShuffleSpecCollection(specs) =>
       specs.exists(isCompatibleWith)
     case _ => false
   }
 
+  def isExpressionsCompatible(other: KeyGroupedShuffleSpec): Boolean = {
```

Comment:
Since this is `KeyGroupedShuffleSpec`, how about `areKeysCompatible`?

Reply:
+1 for @cloud-fan 's suggestion.

Reply:
I'm trying to find the right term for this PR. The term `key` currently is overloaded - it could mean the expressions, or the "partition keys" represented via `partitionValuesOpt`. Maybe I should still call them partition values in a consistent way, and change other places to use "partition values"